src/control/server/server_utils.go

```diff
@@ -205,6 +205,19 @@ func prepBdevStorage(srv *server, iommuEnabled bool, hpiGetter getHugePageInfoFn
 		}
 	}
 
+	for _, engineCfg := range srv.cfg.Engines {
+		// Calculate mem_size per I/O engine (in MB)
+		engineCfg.MemSize = hugePages.Free / len(srv.cfg.Engines)
+		engineCfg.MemSize *= (hugePages.PageSizeKb >> 10)
+		// Pass hugepage size, do not assume 2MB is used
+		engineCfg.HugePageSz = (hugePages.PageSizeKb >> 10)
+		// Warn if hugepages are not enough to sustain average
+		// I/O workload (~1GB)
+		if (engineCfg.MemSize / engineCfg.TargetCount) < 1024 {
+			srv.log.Errorf("Not enough hugepages are allocated!")
+		}
```

Comment:
think it would be good to print hugepage size and memsize to debug using humanize package to get human readable string, could also set local variable for hugePages.PageSizeKb >> 10 as it gets reused here